packages/gestalt/src/BannerOverlay.js

```diff
@@ -0,0 +1,262 @@
+    const messageIsTextNode =
+      typeof message !== 'string' &&
+      Children.only<Element<typeof Text>>(message).type.displayName === 'Text';
+
+    if (messageIsTextNode) {
+      const textColorOverrideStyles = isDarkMode
+        ? styles.textColorOverrideDark
+        : styles.textColorOverrideLight;
+
+      messageTextElement = <span className={textColorOverrideStyles}>{message}</span>;
+    }
+    return messageIsTextNode;
+  };
+
+  const { accessibilityDismissButtonLabel: accessibilityDismissButtonLabelDefault } =
+    useDefaultLabelContext('BannerOverlay');
+
+  const { lightModeBackground, darkModeBackground, textColor } = DEFAULT_COLORS;
+
+  const dismissButtonComponent = !!onDismiss && (
+    <Flex.Item flex="none" alignSelf={isMobileDevice ? 'end' : 'center'}>
+      <IconButton
+        accessibilityLabel={accessibilityDismissButtonLabelDefault}
+        icon="cancel"
+        iconColor="darkGray"
+        onClick={onDismiss}
+        size="xs"
+      />
+    </Flex.Item>
+  );
+  const CTAComponent = primaryAction && (
+    <Flex.Item flex={isMobileDevice ? 'shrink' : 'none'}>
+      {primaryAction.role === 'link' ? (
+        <PrimaryAction
+          accessibilityLabel={primaryAction.accessibilityLabel}
+          href={primaryAction.href}
+          label={primaryAction.label}
+          onClick={primaryAction.onClick}
+          rel={primaryAction?.rel}
+          role="link"
+          size="sm"
+          target={primaryAction?.target}
+        />
+      ) : (
+        <PrimaryAction
+          accessibilityLabel={primaryAction.accessibilityLabel}
+          label={primaryAction.label}
+          onClick={primaryAction.onClick}
+          role="button"
+          size="sm"
+        />
+      )}
+    </Flex.Item>
+  );
+
+  const isMessageTextNode = checkTextNode();
+  return (
+    <Box
+      color={isDarkMode ? darkModeBackground : lightModeBackground}
+      paddingX={4}
+      paddingY={3}
+      rounding={4}
+      borderStyle="shadow"
+      dangerouslySetInlineStyle={
+        isMobileDevice
+          ? {
+              __style: {
+                position: 'fixed',
+                bottom: offset.bottom,
+                left: '50%',
+                transform: 'translateX(-50%)',
+              },
+            }
+          : {
+              __style: {
+                position: 'fixed',
+                top: offset.top,
+                left: '50%',
+                transform: 'translateX(-50%)',
+              },
+            }
+      }
+      position="fixed"
+      display="flex"
+      justifyContent={isMobileDevice ? 'center' : 'between'}
+      alignContent={isMobileDevice ? 'stretch' : 'center'}
+      direction={isMobileDevice ? 'column' : 'row'}
+      smPaddingY={4}
+      fit
+      maxWidth={isMobileDevice ? 348 : 900}
+      width="100%"
+      zIndex={zIndex}
+    >
+      {isMobileDevice && dismissButtonComponent}
+      <Flex alignItems="center" gap={4}>
+        {!!thumbnail?.image &&
```

Comment:
nit: Totally, stylistic, but would it be cleaner to make these a single BannerImage Component?

Then we can also re-use this logic in Toast as a helper

```
<BannerImage thumbnail={thumbnail}>
```

````
const ThumbnailImage = () => {

if(thumbnail.image &&  
Children.only<Element<typeof Image>>(thumbnail.image).type.displayName === 'Image') 
return <ToastImageThumbnail thumbnail>

if(thumbnail.avatar &&  
Children.only<Element<typeof Image>>(thumbnail.image).type.displayName === 'Avatar') 
return <ToastAvatarThumbnail thumbnail>

return null;

}
 
```

Reply:
I guess you can add it here and re-use in both places? 

[packages/gestalt/src/Toast/subcomponents.js](https://github.com/pinterest/gestalt/blob/39211e6b3d6b47a09bb4a840b98dd5147712602f/packages/gestalt/src/Toast/subcomponents.js)

Again, optional

Reply:
Or maybe a new file right next to it

Reply:
I'll create a follow-up for this, since it will also affect Toast